core/eolearn/core/eodata.py

```diff
@@ -279,17 +279,25 @@ def timestamp(self, value: list[dt.datetime]) -> None:
         self.timestamps = value
 
     @property
-    def timestamps(self) -> list[dt.datetime]:
+    def timestamps(self) -> list[dt.datetime] | None:
         """A property for handling the `timestamps` attribute."""
         return self._timestamps
 
     @timestamps.setter
-    def timestamps(self, value: Iterable[dt.datetime]) -> None:
-        if isinstance(value, Iterable) and all(isinstance(time, (dt.date, str)) for time in value):
+    def timestamps(self, value: Iterable[dt.datetime] | None) -> None:
+        if value is None:
+            self._timestamps = None
+        elif isinstance(value, Iterable) and all(isinstance(time, (dt.date, str)) for time in value):
             self._timestamps = [parse_time(time, force_datetime=True) for time in value]
         else:
             raise TypeError(f"Cannot assign {value} as timestamps. Should be a sequence of datetime.datetime objects.")
 
+    def get_timestamps(self, message_on_failure: str = "The EOPatch does not contain timestamps.") -> list[dt.datetime]:
```

Comment:
I understand the need for this function is to not change all the types and interactions of the timestamps object with the rest of the code in the repo. 

However, I see some possible issues:
- people would use `get_timestamps`, thinking it's the same as `eop.timestamps`
- people would use `eop.timestamps` and not knowing about this function, they would implement their own versions on how to handle `None` cases

I have no idea if there is any way how to minimize the chances of the above happening, at least more than what you already documented. 

So I guess this is more a comment than anything else, just trying to see if you had similar concerns, but probably there is nothing else to do here but to let developers find this out on their own, or correct them during PRs.

Reply:
My first thought was to not provide this at all. But then you really do need to do this in a lot of places, so I decided to add this with a way to provide a custom message anyway. I believe the documentation is sufficient to address your first bullet point. As for the second one, no clue how to do that :/

Reply:
Yeah, let's see how it turns out, not much else we can do at the moment without changing a bunch of code.

Unless you think changing the code is the proper way to do it, then we could address this some other time and open a ticket for handling this

Reply:
i dont really see any other way of doing it, but i'm open to suggestions. It has also been raised in the ticket since last week with 0 replies :)

Reply:
You're right, I guess without this users would implement their own None handling, which would then become a pattern converging to the same thing. 

I tried to think if better names could be found, but I couldn't think of any, so let's just go ahead with it as it is.

Reply:
i was thinking about `get_timestamps_or_fail` to be more explicit in what the intended use is, but it is a bit verbose